src/plugins/controls/public/control_group/editor/control_group_editor.tsx

```diff
@@ -28,38 +30,101 @@ import {
   EuiButtonEmpty,
   EuiSpacer,
   EuiCheckbox,
+  EuiForm,
+  EuiAccordion,
+  useGeneratedHtmlId,
+  EuiSwitch,
+  EuiText,
+  EuiHorizontalRule,
 } from '@elastic/eui';
 
+import { CONTROL_LAYOUT_OPTIONS, CONTROL_WIDTH_OPTIONS } from './editor_constants';
 import { ControlGroupStrings } from '../control_group_strings';
 import { ControlStyle, ControlWidth } from '../../types';
-import { CONTROL_LAYOUT_OPTIONS, CONTROL_WIDTH_OPTIONS } from './editor_constants';
+import { ParentIgnoreSettings } from '../..';
+import { ControlsPanels } from '../types';
+import { ControlGroupInput } from '..';
+import {
+  DEFAULT_CONTROL_WIDTH,
+  getDefaultControlGroupInput,
+} from '../../../common/control_group/control_group_constants';
 
 interface EditControlGroupProps {
-  width: ControlWidth;
-  controlStyle: ControlStyle;
-  setAllWidths: boolean;
+  initialInput: ControlGroupInput;
   controlCount: number;
-  updateControlStyle: (controlStyle: ControlStyle) => void;
-  updateWidth: (newWidth: ControlWidth) => void;
-  updateAllControlWidths: (newWidth: ControlWidth) => void;
-  onCancel: () => void;
+  updateInput: (input: Partial<ControlGroupInput>) => void;
+  onDeleteAll: () => void;
   onClose: () => void;
 }
 
+type EditorControlGroupInput = ControlGroupInput &
+  Required<Pick<ControlGroupInput, 'defaultControlWidth'>>;
+
+const editorControlGroupInputIsEqual = (a: ControlGroupInput, b: ControlGroupInput) =>
+  fastIsEqual(a, b);
+
 export const ControlGroupEditor = ({
-  width,
-  controlStyle,
-  setAllWidths,
   controlCount,
-  updateControlStyle,
-  updateWidth,
-  updateAllControlWidths,
-  onCancel,
+  initialInput,
+  updateInput,
+  onDeleteAll,
   onClose,
 }: EditControlGroupProps) => {
-  const [currentControlStyle, setCurrentControlStyle] = useState(controlStyle);
-  const [currentWidth, setCurrentWidth] = useState(width);
-  const [applyToAll, setApplyToAll] = useState(setAllWidths);
+  const [resetAllWidths, setResetAllWidths] = useState(false);
```

Comment:
Even with it defaulting to false, this is still such a strange behaviour to me. I figure we keep it for now because it does add a slightly easier way to change all widths, but having a checkbox that is applied one-time on save just seems weird.